controllers/imageclusterinstall_controller.go

```diff
@@ -537,41 +554,101 @@ func (r *ImageClusterInstallReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Complete(r)
 }
 
-func (r *ImageClusterInstallReconciler) setBMHImage(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost, url string) error {
+func (r *ImageClusterInstallReconciler) getBmhDataImage(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost) *bmh_v1alpha1.DataImage {
+	if bmh == nil {
+		return nil
+	}
+	dataImage := bmh_v1alpha1.DataImage{}
+	key := client.ObjectKey{
+		Name:      bmh.Name,
+		Namespace: bmh.Namespace,
+	}
+	err := r.Get(ctx, key, &dataImage)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+
+	return &dataImage
+}
+
+func (r *ImageClusterInstallReconciler) isInspectionEnabled(bmh *bmh_v1alpha1.BareMetalHost) bool {
+	if bmh == nil {
+		return false
+	}
+
+	if bmh.ObjectMeta.Annotations != nil && bmh.ObjectMeta.Annotations[inspectAnnotation] != "disabled" {
+		return true
+	}
+
+	return false
+}
+
+func (r *ImageClusterInstallReconciler) createBMHDataImage(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost, ici *v1alpha1.ImageClusterInstall, url string) error {
+	dataImage := r.getBmhDataImage(ctx, bmh)
+
+	if dataImage == nil {
+		r.Log.Infof("creating new dataImage for BareMetalHost (%s/%s)", bmh.Name, bmh.Namespace)
+
+		// Name and namespace must match the ones in BMH
+		dataImage = &bmh_v1alpha1.DataImage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      bmh.Name,
+				Namespace: bmh.Namespace,
+			},
+			Spec: bmh_v1alpha1.DataImageSpec{
+				URL: url,
+			},
+		}
+		err := controllerutil.SetControllerReference(bmh, dataImage, r.Client.Scheme())
+		if err != nil {
+			return fmt.Errorf("failed to set controller reference for dataImage due to %w", err)
+		}
+
+		err = r.Create(ctx, dataImage)
```

Comment:
Log that we created the data image (add namespace + name in the log)

Reply:
We do, see few lines above
```go
r.Log.Infof("creating new dataImage for BareMetalHost (%s/%s)", bmh.Name, bmh.Namespace)
```